### PR TITLE
avocado.core.loader fix crash on loader exception

### PR DESCRIPTION
**Pull request by apahim**
v1:
 - Fake the test class when the original class cannot be loaded.
 - Capture the load exception and log it.
 - Set `status` = `ERROR` to the test that failed to be loaded.

### CODE REVIEW

**Comment by apahim** [comment]
@ldoktor, @clebergnu, how much do you like this pr?

**Comment by ldoktor** [comment]
I'm sorry, I saw it couple of times but always got distracted. One thing it'd be nice to add selftest for this...

File: avocado/core/test.py [hunk 1]
```diff
@@ -777,3 +777,17 @@ class ReplaySkipTest(SkipTest):
...
+    Handle when test loader faces an exception (i.e. python code error).
+    """
+
+    def __init__(self, *args, **kwargs):
+        exception = kwargs.pop('exception')
+        Test.__init__(self, *args, **kwargs)
+        self.exception = exception
+
+    def test(self):
+        raise exceptions.TestLoaderException(self.exception)
```
Comment (ldoktor) [inline avocado/core/test.py]: I don't see much benefit in having extra exception class for this kind of failure. Pure `self.fail(...)` would be IMO sufficient.

File: avocado/core/runner.py [hunk 2]
```diff
@@ -233,21 +228,13 @@ def _run_test(self, test_factory, queue):
...
 
-        try:
-            instance = loader.load_test(test_factory)
-            if instance.runner_queue is None:
-                instance.runner_queue = queue
-            runtime.CURRENT_TEST = instance
-            early_state = instance.get_state()
-            early_state['early_status'] = True
-            queue.put(early_state)
-        except Exception:
```
Comment (ldoktor) [inline avocado/core/runner.py]: Well this was here for a reason. When anything here fails, avocado hangs. It's true that we added 10s deadline, so maybe it's time to say good buy to this, but I'd prefer keep it here (and remove the `Exception` to catch really everything).

Comment (ldoktor) [inline avocado/core/runner.py]: Actually I thought about it and it's a corner case, the main thread should cope with that (using the 10s deadline) so yes, it's probably time to remove this hack and simply run the test. (the simpler - the better)

File: avocado/core/test.py (hunk 1 above)
Comment (ldoktor) [inline avocado/core/test.py]: Actually `self.error`...

Comment (apahim) [inline avocado/core/test.py]: ok
